### PR TITLE
feat(windrose): Modifying windrose for GH component.

### PR DESCRIPTION
**Pull request by saeranv**
@chriswmackey 
cc @mostaphaRoudsari 

These are improvements to the wind rose module that the new GH component relies on. 

Changes:
- Adding more settable properties: north, different frequency controls
- Change frequency calculation to be another histogram calculation 
- Made defaults more explicit
- More detailed doc strings
- More tests
- General code clean-up/corrections

### CODE REVIEW

File: ladybug/windrose.py [hunk 1]
```diff
@@ -188,19 +232,37 @@ def legend_parameters(self):
...
 
+    @property
+    def north(self):
+        """North orientation for windrose by degrees in counterclockwise orienation.
+
+        This must be a number between -360 and 360. 0 is North, 90 is West and 270
+        is East.
+        """
+        if self._north is None:
+            self._north = self.DEFAULT_NORTH
```
Comment (chriswmackey) [inline ladybug/windrose.py:254]: Why not just `return self.DEFAULT_NORTH`? Here you are loosing track of the fact that the user did not specify this value for `_north`. Or, if this information about whether the north was user-assigned is not important, why bother setting `_north` to `None` in the first place.

File: ladybug/windrose.py [hunk 2]
```diff
@@ -188,19 +232,37 @@ def legend_parameters(self):
...
+        return self._north
+
+    @north.setter
+    def north(self, north):
+        assert isinstance(north, float), 'north must be a float representing degrees.' \
+            ' Got {}.'.format(north)
+
+        self._compass = None
+        self._container = None
+        self._north = north % 360.0
```
Comment (chriswmackey) [inline ladybug/windrose.py:264]: Why do you divide by 360 here? Is the `_north` not in degrees?

Comment (saeranv) [reply ladybug/windrose.py:264]: Yes this is in degrees. But it's not a slash, it's a modulo sign (%). It's just there to reset values > 360 back to 0.

File: ladybug/windrose.py (hunk 1 above)
Comment (saeranv) [reply ladybug/windrose.py:254]: I see your point, it doesn't have to be set to None in the first place. I'll set it to the default value in the __init__ function.

File: ladybug/windrose.py (hunk 2 above)
Comment (chriswmackey) [reply ladybug/windrose.py:262]: Ah, that's actually very elegant. Sorry that I forgot the modulo is the remainder of the division and not the result of the division itself.

**Comment by chriswmackey** [comment]
Thanks @saeranv . I am going to merge this now.

**Pull request merged**